lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb

```diff
@@ -91,6 +91,9 @@ def get_upload_token(app_id, binary_path)
       def upload_binary(app_id, binary_path)
         connection.post(binary_upload_url(app_id), File.open(binary_path).read) do |request|
           request.headers["Authorization"] = "Bearer " + @auth_token
+          request.headers["X-APP-DISTRO-API-CLIENT-ID"] = "fastlane"
+          request.headers["X-APP-DISTRO-API-CLIENT-TYPE"] =  Actions.lane_context[Actions::SharedValues::PLATFORM_NAME].to_s
```

Comment:
We did consider making platform a class level variable in action, but the assignment of class level variable during runtime does not seem to be feasible. We had also thought about passing in platform as an instance variable or parameters, but since it is only used once looking for the platform name again didn't seem too expensive.

Reply:
I thought "client type" was supposed to be the type of client, i.e. gradle, Firebase CLI, etc. Shouldn't "fastlane" go here instead of the platform?

Reply:
That is client id. Client type is the android vs ios. I can link you to the developer API docs.

Reply:
I see. That conflicts with the Gradle plugin though, AFAICT: https://github.com/FirebasePrivate/firebase-android-buildtools/blob/a4821b09655eebf1593ac32e38c52ad8c9f47afb/firebase-appdistribution/appdistribution-gradle/src/main/java/com/google/firebase/appdistribution/CliRuntimeProperties.java#L23

@tagboola is it possible the docs are wrong: https://docs.google.com/document/d/16wkaDtE0W2dTjfn_dvSGDPRbpTOSjAzhFf_IyARQGwc/edit#heading=h.msoyza6jz9fg

Reply:
@lfkellogg I wrote the docs so there is DEFINITELY a possibility that they are wrong 😆 

But in this case setting the client type to ios or android is correct. If you look at code in the Firebase CLI we set the `X-APP-DISTRO-API-CLIENT-TYPE` header  to platform:
https://github.com/firebase/firebase-tools/blob/57d008a5eebc9f091e78c07134ce2625a4d13156/src/appdistribution/client.ts#L62

For some reason in the gradle plugin we set the `X-APP-DISTRO-API-CLIENT-TYPE` to gradle, but I assume the dashboard is smart enough to know that gradle means android.
https://github.com/FirebasePrivate/firebase-android-buildtools/blob/98711268db98522c118585f0ab6728e9ec1a5a38/firebase-appdistribution/appdistribution-gradle/src/main/groovy/com/google/firebase/appdistribution/gradle/UploadDistributionTask.groovy#L117

I think I would prefer this be passed as an instance variable to the api client. Even though it's only used once the client depends on it so I think it's fine to pass it in.

Reply:
OK sounds good, thanks guys!